### ISSUE TITLE
"csc.exe" exited with code -2146232797

### ISSUE THREAD

**Issue by dely-ncci** [dotnet/roslyn#10885]
**Version Used**:  1.2.0.60317

**Steps to Reproduce**:

```
using System.Collections.Generic;
using System.Linq;
namespace ConsoleApplication1
{
    class Program
    {
        public class AClass : List<string>
        {
        }
        static void Main(string[] args)
        {
            var b = (from a in AClass select a);
        }
    }
}
```

**Expected Behavior**:

Error message 

**Actual Behavior**:

Problem signature:
  Problem Event Name:	CLR20r3
  Problem Signature 01:	VBCSCompiler.exe
  Problem Signature 02:	1.2.0.60317
  Problem Signature 03:	d90b2aed
  Problem Signature 04:	Microsoft.CodeAnalysis.CSharp
  Problem Signature 05:	1.2.0.60317
  Problem Signature 06:	a2eab45f
  Problem Signature 07:	5e30
  Problem Signature 08:	3f2
  Problem Signature 09:	System.InvalidOperationException
  OS Version:	6.1.7601.2.1.0.256.4
  Locale ID:	1033
  Additional Information 1:	f3b8
  Additional Information 2:	f3b85aa92f370ccea20162cf3275b897
  Additional Information 3:	38ec
  Additional Information 4:	38ec5120b1c4325d71d35defcb892ddc

**Comment by leppie** [comment]
Reproduced, nice catch :D

**Comment by alrz** [comment]
Changing this to BindValue solves the problem. I don't know in what case its Color Color.

http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Binder/Binder_Query.cs,24

**Comment by leppie** [comment]
Note the problem is reproduced even when `AClass` is on namespace level, so I am not sure why it is even being considered for the `ColorColor` case.

**Comment by gafter** [comment]
This was fixed in #11686.
